### PR TITLE
Improve description of `Callable.bind/unbind`

### LINKED ISSUE #76141: Callable unbind doesn't actually unbind arguments
### Godot version

4.0.2

### System information

Linux 5.10.167-2-MANJARO

### Issue description

Calling `unbind` on a callable doesn't actually unbind the arguments. However it makes godot think the arguments are unbound, thus requiring you to call the callable with fewer arguments. But some of the arguments passed along in the call will be ignored, and the original bound arguments are used instead.

### Steps to reproduce

Run
```gdscript
func foo(a: int, b: int, c: int):
	print("a: ", a)
	print("b: ", b)
	print("c: ", c)

func _ready():
	var callable = foo.bind(1,2)
	callable = callable.unbind(1)
	callable.call(4,5)
	# errors
	# callable.call(4)
```

The printed output is:
```
a: 4
b: 1
c: 2
```
Whereas id expect to see something like
```
a: 4
b: 5
c: 1
```

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by AThousandShips**
Fixes #76141

### CODE REVIEW

**Comment by timothyqiu** [comment]
> [b]Note:[/b] This does not remove arguments bound with [method bind].

Technically it does remove arguments bound with `bind()`.

```gdscript
func _ready() -> void:
	# the 42 bound by `bind(42)` is unbound by `unbind(1)`.
	var f := foo.unbind(1).bind(42)
	# so you have to provide exactly one argument here.
	f.call(100)

func foo(a: int) -> void:
	print(a)
```

The key is that `bind()` and `unbind()` are decorators for the argument list provided to its own `call()` method, producing a new callable. So they are applied backwards when chained.

```gdscript
foo.unbind(1).bind(42).call(100)

call(100) #      argument list is: 100
bind(42)  #             append 42: 100, 42
unbind(1) # remove 1 from the end: 100

foo.bind(88, 99).unbind(1).call(22, 33)

call(22, 33) #      argument list is: 22, 33
unbind(1)    # remove 1 from the end: 22
bind(88, 99) #         append 88, 99: 22, 88, 99
```

**Comment by AThousandShips** [comment]
Will add that it doesn't remove arguments *previously* bound for clarification

**Comment by AThousandShips** [comment]
Clarified that description and added an example to illustrate it.

File: doc/classes/Callable.xml [hunk 1]
```diff
@@ -187,7 +187,13 @@
 			<return type="Callable" />
 			<param index="0" name="argcount" type="int" />
 			<description>
-				Returns a copy of this [Callable] with the arguments unbound, as defined by [param argcount]. Calling the returned [Callable] will call the method without the extra arguments that are supplied in the [Callable] on which you are calling this method.
+				Returns a copy of this [Callable] with the arguments unbound, as defined by [param argcount]. Calling the returned [Callable] will call the method while ignoring the last [param argcount] arguments supplied in that call, in other words it will call the method as if it was called with [param argcount] fewer arguments.
```
Comment (YuriSizov) [inline doc/classes/Callable.xml:190]:
```suggestion
				Returns a copy of this [Callable] with a number of arguments unbound. In other words, when the new callable is called the last few arguments supplied by the user are ignored, according to [param argcount]. The remaining arguments are passed to the callable. This allows to use the original callable in a context that attempts to pass more arguments than this callable can handle, e.g. a signal with a fixed number of arguments.
```

Comment (AThousandShips) [reply doc/classes/Callable.xml]: Thank you that is much more clear and consice

File: doc/classes/Callable.xml [hunk 2]
```diff
@@ -187,7 +187,13 @@
 			<return type="Callable" />
 			<param index="0" name="argcount" type="int" />
 			<description>
-				Returns a copy of this [Callable] with the arguments unbound, as defined by [param argcount]. Calling the returned [Callable] will call the method without the extra arguments that are supplied in the [Callable] on which you are calling this method.
+				Returns a copy of this [Callable] with a number of arguments unbound. In other words, when the new callable is called the last few arguments supplied by the user are ignored, according to [param argcount]. The remaining arguments are passed to the callable. This allows to use the original callable in a context that attempts to pass more arguments than this callable can handle, e.g. a signal with a fixed number of arguments.
+				[b]Note:[/b] This does not remove arguments previously bound with [method bind].
```
Comment (YuriSizov) [inline doc/classes/Callable.xml:191]:
```suggestion
				[b]Note:[/b] When this method is chained with other similar method, the order in which the argument list is modified is read from the right.
```

File: doc/classes/Callable.xml [hunk 3]
```diff
@@ -187,7 +187,13 @@
 			<return type="Callable" />
 			<param index="0" name="argcount" type="int" />
 			<description>
-				Returns a copy of this [Callable] with the arguments unbound, as defined by [param argcount]. Calling the returned [Callable] will call the method without the extra arguments that are supplied in the [Callable] on which you are calling this method.
+				Returns a copy of this [Callable] with a number of arguments unbound. In other words, when the new callable is called the last few arguments supplied by the user are ignored, according to [param argcount]. The remaining arguments are passed to the callable. This allows to use the original callable in a context that attempts to pass more arguments than this callable can handle, e.g. a signal with a fixed number of arguments.
```
Comment (YuriSizov) [inline doc/classes/Callable.xml:190]:
```suggestion
				Returns a copy of this [Callable] with a number of arguments unbound. In other words, when the new callable is called the last few arguments supplied by the user are ignored, according to [param argcount]. The remaining arguments are passed to the callable. This allows to use the original callable in a context that attempts to pass more arguments than this callable can handle, e.g. a signal with a fixed number of arguments. See also [method bind].
```

File: doc/classes/Callable.xml (hunk 2 above)
Comment (YuriSizov) [reply doc/classes/Callable.xml:191]: This note can also be added to bind and bindv.

File: doc/classes/Callable.xml (hunk 3 above)
Comment (YuriSizov) [reply doc/classes/Callable.xml:190]: Didn't manage to edit my previous suggestion before you applied it :P

Comment (AThousandShips) [reply doc/classes/Callable.xml:190]: My bad! Anything else before I apply these?

Comment (YuriSizov) [reply doc/classes/Callable.xml:190]: No, it's my fault, I didn't use the review function. I don't have further suggestions at the moment.

File: doc/classes/Callable.xml [hunk 4]
```diff
@@ -187,7 +189,13 @@
 			<return type="Callable" />
 			<param index="0" name="argcount" type="int" />
 			<description>
-				Returns a copy of this [Callable] with the arguments unbound, as defined by [param argcount]. Calling the returned [Callable] will call the method without the extra arguments that are supplied in the [Callable] on which you are calling this method.
+				Returns a copy of this [Callable] with a number of arguments unbound. In other words, when the new callable is called the last few arguments supplied by the user are ignored, according to [param argcount]. The remaining arguments are passed to the callable. This allows to use the original callable in a context that attempts to pass more arguments than this callable can handle, e.g. a signal with a fixed number of arguments. See also [method bind].
+				[b]Note:[/b] When this method is chained with other similar methods binds from later calls appear before earlier calls.
```
Comment (YuriSizov) [inline doc/classes/Callable.xml:193]: I don't think it's clear what's later and what's earlier. Which is why I suggested to explicitly say "read from the right". And your following example demonstrates how to read it from the right exactly.

Comment (AThousandShips) [reply doc/classes/Callable.xml:193]: True, using "right to left" to make it more clear what right means

**Review by YuriSizov — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by AThousandShips** [comment]
Thank you!

**Pull request merged**